src/LagrangianParticleTracking/update_particle_properties.jl

```diff
@@ -48,37 +121,38 @@ function update_particle_properties!(lagrangian_particles, model, Δt)
     workgroup = min(length(lagrangian_particles), 256)
     worksize = length(lagrangian_particles)
 
+    arch = architecture(model.grid)
```

Comment:
Totally trivial but 

```suggestion
    arch = architecture(model)
```

is better (for models that don't have `.grid`)